drivers/gpio/gpio_stm32.c

```diff
@@ -525,28 +568,11 @@ static const struct gpio_driver_api gpio_stm32_driver = {
  */
 static int gpio_stm32_init(const struct device *device)
 {
-	const struct gpio_stm32_config *cfg = device->config;
 	struct gpio_stm32_data *data = device->data;
 
 	data->dev = device;
 
-	/* enable clock for subsystem */
-	const struct device *clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
-
-	if (clock_control_on(clk,
-			     (clock_control_subsys_t *)&cfg->pclken) != 0) {
-		return -EIO;
-	}
-
-#if defined(PWR_CR2_IOSV) && DT_NODE_HAS_STATUS(DT_NODELABEL(gpiog), okay)
-	if (cfg->port == STM32_PORTG) {
-		/* Port G[15:2] requires external power supply */
-		/* Cf: L4/L5 RM, Chapter "Independent I/O supply rail" */
-		z_stm32_hsem_lock(CFG_HW_RCC_SEMID, HSEM_LOCK_DEFAULT_RETRY);
-		LL_PWR_EnableVddIO2();
-		z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
-	}
-#endif  /* PWR_CR2_IOSV */
+	gpio_stm32_clock_request(device, true);
 
 	return 0;
```

Comment:
could be `return gpio_stm32_clock_request()` ?

Reply:
Indeed, fixed.